scan/lib/scan/detect_values.rb

```diff
@@ -100,6 +108,41 @@ def self.filter_simulators(simulators, operator = :greater_than_or_equal, deploy
       end
     end
 
+    def self.detect_sdk_version(platform)
+      UI.crash!("Unknown platform: #{platform}") unless PLATFORM_SIMULATOR_NAME.key?(platform)
+      simulator_name = PLATFORM_SIMULATOR_NAME[platform]
+      @sdk_versions ||= {}
+      @sdk_versions[platform] ||= begin
+         platform_path = Pathname.new("Platforms/#{simulator_name}.platform/Developer/SDKs/")
+         sdks_path = Pathname.new(FastlaneCore::Helper.xcode_path).join(platform_path)
+         default_sdk_path = sdks_path.join("#{simulator_name}.sdk")
+         sdk_path = sdks_path.children.find { |child| child.symlink? && child.realpath == default_sdk_path }
+         UI.crash!("Unable to find default #{simulator_name} SDK version from SDKs: #{sdks_path.children}") unless sdk_path
+
+         version = /#{Regexp.quote(simulator_name)}(?<val>.*)\.sdk/.match(sdk_path.basename.to_s)
+         UI.crash!("Could not determine SDK version from #{sdk_path}") unless version && version[:val] != ''
+
+         begin
+           Gem::Version.new(version[:val])
+         rescue ArgumentError => e
+           UI.crash!("Could not parse SDK version: #{e.message}")
+         end
+       end
+    end
```

Comment:
Sounds good, certainly feels more robust to use CLI output rather than relying on directory structure. Will work on a revision to use this approach instead. Thanks!

Reply:
Thanks for the feedback and the sample code! I've rewritten the implementation to invoke the suggested commands.